### PR TITLE
Read URL constructor from global if supported

### PR DESCRIPTION
**Pull request by roderickhsiao**
Address issue https://github.com/TooTallNate/spotify-uri/issues/24

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,10 +4,10 @@
     "module": "CommonJS",
     "target": "es2015",
     "esModuleInterop": true,
-    "lib": ["esnext"],
+    "lib": ["esnext", "dom"],
```
Comment (Kikobeats) [inline tsconfig.json:7]: is this change actually necessary?

Comment (roderickhsiao) [reply tsconfig.json:7]: yeah, Typescript is complaining direct access of `window`, even if we are doing `typeof window !== 'undefined`

**Comment by Kikobeats** [comment]
I'm still thinking if this make sense in terms of maintaining the project in the long-term.

You can easily have the same behavior a bundler ([example](https://github.com/microlinkhq/mql/blob/master/rollup.config.js#L22=)).

In any case, I would accept `window` typecheck, but `global`is too dangerous, since anyone can mutate the global scope and the library definetively shouldn't trust in global things

**Comment by roderickhsiao** [comment]
Make sense to me, I'll close this PR then! 

Thanks for being patient with me
